machine-config/harvester.vue

```diff
@@ -0,0 +1,446 @@
+
+        (res.configMaps.value || []).map((O) => {
+          if (O.isHciUserConfig) {
+            userDataOptions.push({
+              label: O.metadata.name,
+              value: O.data.cloudInit
+            });
+          }
+
+          if (O.isHciNetworkConfig) {
+            networkDataOptions.push({
+              label: O.metadata.name,
+              value: O.data.cloudInit
+            });
+          }
+        });
+
+        this.userDataOptions = userDataOptions;
+        this.networkDataOptions = networkDataOptions;
+
+        this.imageOptions = (res.images.value || []).filter( (O) => {
+          return !O.spec.url.endsWith('.iso');
+        }).map( (O) => {
+          const value = O.id;
+          const label = `${ O.spec.displayName } (${ value })`;
+
+          return {
+            label,
+            value
+          };
+        });
+
+        this.networkOptions = (res.networks.value || []).map( (O) => {
+          let value;
+          let label;
+
+          try {
+            const config = JSON.parse(O.spec.config);
+
+            const id = config.vlan;
+
+            value = O.id;
+            label = `${ value } (vlanId=${ id })`;
+          } catch (err) {}
+
+          return {
+            label,
+            value
+          };
+        });
+
+        this.namespaceOptions = (res.namespaces.value || [])
+          .filter(namespace => !namespace.isSystem)
+          .map( (namespace) => {
+            const value = namespace.metadata.name;
+            const label = namespace.metadata.name;
+
+            return {
+              label,
+              value
+            };
+          });
+      }
+
+      if (isEmpty(this.value.cpuCount)) {
+        this.value.cpuCount = '2';
+      }
+
+      if (isEmpty(this.value.memorySize)) {
+        this.value.memorySize = '4';
+      }
+
+      if (isEmpty(this.value.diskSize)) {
+        this.value.diskSize = '40';
+      }
+    } catch (e) {
+      this.errors = exceptionToErrorsArray(e);
+    }
+  },
+
+  data() {
+    return {
+      credential:         null,
+      isImportCluster:    false,
+      cloudConfig:        '',
+      networkData:        '',
+      imageOptions:       [],
+      namespaceOptions:   [],
+      networkOptions:     [],
+      userDataOptions:    [],
+      networkDataOptions: [],
+      cpuCount:           ''
+    };
+  },
+
+  computed: { ...mapGetters({ t: 'i18n/t' }) },
+
+  watch: {
+    'credentialId'() {
+      this.imageOptions = [];
+      this.networkOptions = [];
+      this.namespaceOptions = [];
+      this.value.imageName = '';
+      this.value.networkName = '';
+      this.value.vmNamespace = '';
+
+      this.$fetch();
+    },
+
+    networkData(neu) {
+      this.value.networkData = neu;
+    },
+
+    cloudConfig(neu) {
+      this.value.cloudConfig = neu;
+    },
+  },
+
+  methods: {
+    stringify,
+
+    test() {
+      const errors = [];
+
+      if (!this.value.cpuCount || this.value.cpuCount === 'null') {
```

Comment:
Why are these values ever the _string_ `'null'`?

Reply:
When using the UnitInput component to return a string (outputAs = 'String'), I now add a layer of judgment to the UnitInput component that returns an empty string if the value returned is null
```
if ( this.outputAs === 'string' ) {
      out = out === null ? '' : `${ out }`;
    }
```